server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1967,6 +1967,12 @@ public void addFlushListener(Translog.Location location, ActionListener<Long> li
         listener.onFailure(new UnsupportedOperationException("Engine type " + this.getClass() + " does not support flush listeners."));
     }
 
+    public void addGenerationDurabilityListener(long generation, ActionListener<Void> listener) {
```

Comment:
Naming-wise, I wonder if we can have flush in the name, perhaps `addFlushDurabilityListener`?

Can we add javadoc on the method? I think the contract is that you add the listener and then it is automatically "removed" once it is invoked?